### PR TITLE
docs: add encryption infos for self-hosting

### PR DESCRIPTION
**Pull request by Shegox**
## Changes:

This PR adds some clarification about the encryption usage in a self-hosted setup. For this it adds 
- to the `privateKey` config the information on how to encrypt a value using openssl
- A note that decryption is only applied to repository/user configs and can't be used in the admin config.

## Context:

While trying to get encryption working, I run into the problem that I tried to add the encrypted value to the admin config, right next to the privateKey (kinda pointless, but yeah) and noticed that the value is not decrypted from the adminconfig (makes sense), but took me a couple tries to figure that out. Therefore I'm adding that to the docs. (kinda related to https://github.com/renovatebot/config-help/issues/848#issuecomment-688293992) 

The second thing is to add the handy command to encrypt a value directly from the command line, which is for trying out quicker than adjusting the encryption page with your own `privateKey`.

 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -330,6 +330,10 @@ To create the key pair with OpenSSL use the following commands:
 - `openssl genrsa -out rsa_priv.pem 4096` for generating the private key
 - `openssl rsa -pubout -in rsa_priv.pem -out rsa_pub.pem` for extracting the public key
 
+To encrypt a secret using openssl use following command:
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:333]:
```suggestion
To encrypt a secret with OpenSSL use the following command:
```

File: docs/usage/self-hosted-configuration.md [hunk 2]
```diff
@@ -330,6 +330,10 @@ To create the key pair with OpenSSL use the following commands:
...
+
+- `echo 'your-secret' | openssl rsautl -encrypt -pubin -inkey rsa_pub.pem | base64`, replacing `your-secret` with your actual secret
+
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:336]:
```bash
echo 'your-secret' | openssl rsautl -encrypt -pubin -inkey rsa_pub.pem | base64
```

Replace `your-secret` with your actual secret.

---

Please make it look like the section above.
I don't know how to suggest this in a way that GitHub will accept unfortunately.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/self-hosted-configuration.md (hunk 2 above)
Comment (Shegox) [reply docs/usage/self-hosted-configuration.md]: Thanks, added it. 
> Please make it look like the section above.
Or do you mean I should add it just as a bullet point to the two commands above?

Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md]: Thank you, but I wasn't clear enough on the change I wanted you to make... 😉 

Please put the code in a code block, like this:

```bash
echo 'your-secret' | openssl rsautl -encrypt -pubin -inkey rsa_pub.pem | base64
```

To do this: 

1. Remove bullet point `-` character.
2. Wrap your code into a code block (3 backticks, code, 3 backticks).
3. Put the comment about what to replace below the codeblock.

https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

Comment (Shegox) [reply docs/usage/self-hosted-configuration.md]: Got you now, thanks for the clarification. 
Incorporated it now 👍

Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md]: We got there in the end! 😄 🎉 ✨

**Review by HonkingGoose — COMMENTED** [review]
Docs styling/grammar is okay now.

I'll let somebody else review the command that this PR adds.

File: docs/usage/private-modules.md [hunk 3]
```diff
@@ -212,6 +212,8 @@ For instructions on this, see the above section on encrypting secrets for the Wh
 - Use the resulting HTML encrypt page to encrypt secrets for your app before adding them to user/repository config
 - Configure the app to run with `privateKey` set to the private key you generated above
 
+Note: Encrypted values can't be used in the "AdminBot config".
```
Comment (rarkins) [inline docs/usage/private-modules.md:215]: "admin config"

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/private-modules.md (hunk 3 above)
Comment (Shegox) [reply docs/usage/private-modules.md]: I used the spelling from above, however now updated in both places from `AdminBot config` to `Admin/Bot config`

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
Thank you @Shegox !

**Review by viceice — APPROVED** [review]
❤️

**Pull request merged**